src/lib/certmap/sss_cert_content_crypto.c

```diff
@@ -484,6 +484,69 @@ static int add_oid_to_san_list(TALLOC_CTX *mem_ctx,
     return 0;
 }
 
+/* Due to CVE-2023-0286 the type of the x400Address member of the
+ * GENERAL_NAME struct was changed from ASN1_TYPE to ASN1_STRING. The
+ * following code tries to make sure that the x400Address can be extracted from
+ * the certificate in either case. */
+static int get_x400address_data(TALLOC_CTX *mem_ctx, GENERAL_NAME *current,
+                                unsigned char **_data, int *_len)
+{
+    int ret;
+    unsigned char *data = NULL;
+    int len;
+
+#ifdef HAVE_X400ADDRESS_STRING
+    len = ASN1_STRING_length(current->d.x400Address);
+    if (len <= 0) {
+        ret = EINVAL;
+        goto done;
+    }
+
+    data = (unsigned char *) talloc_strndup(mem_ctx,
+                   (const char *) ASN1_STRING_get0_data(current->d.x400Address),
+                   len);
```

Comment:
Is there a guarantee that resulting string isn't less than `len` (i.e. source string doesn't contain '\0' inside)?

Reply:
I added a `strlen()` to be on the safe side.